exchanges/bittrex/bittrex_websocket.go

```diff
@@ -0,0 +1,584 @@
+						b.Name,
+						err)
+				}
+			}
+		case "ticker":
+			for j := range response.Message[i].Arguments {
+				var tickerUpdate TickerData
+				err = b.wsDecodeMessage(response.Message[i].Arguments[j], &tickerUpdate)
+				if err != nil {
+					return err
+				}
+				err = b.WsProcessUpdateTicker(tickerUpdate)
+				if err != nil {
+					return err
+				}
+			}
+		case "marketSummary":
+			for j := range response.Message[i].Arguments {
+				var marketSummaryUpdate MarketSummaryData
+				err = b.wsDecodeMessage(response.Message[i].Arguments[j], &marketSummaryUpdate)
+				if err != nil {
+					return err
+				}
+
+				err = b.WsProcessUpdateMarketSummary(&marketSummaryUpdate)
+				if err != nil {
+					return err
+				}
+			}
+		case "heartbeat":
+			if b.Verbose {
+				log.Warnf(log.WebsocketMgr, "%s Received heartbeat\n", b.Name)
+			}
+		case "authenticationExpiring":
+			if b.Verbose {
+				log.Debugf(log.WebsocketMgr, "%s - Re-authenticating.\n", b.Name)
+			}
+			err = b.WsAuth()
+			if err != nil {
+				b.Websocket.DataHandler <- err
+				b.Websocket.SetCanUseAuthenticatedEndpoints(false)
+			}
+		case "order":
+			for j := range response.Message[i].Arguments {
+				var orderUpdate OrderUpdateMessage
+				err = b.wsDecodeMessage(response.Message[i].Arguments[j], &orderUpdate)
+				if err != nil {
+					return err
+				}
+				err = b.WsProcessUpdateOrder(&orderUpdate)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// WsProcessUpdateTicker processes an update on the ticker
+func (b *Bittrex) WsProcessUpdateTicker(tickerData TickerData) error {
+	pair, err := currency.NewPairFromString(tickerData.Symbol)
+	if err != nil {
+		return err
+	}
+
+	tickerPrice, err := ticker.GetTicker(b.Name, pair, asset.Spot)
+	if err != nil {
+		// Received partial data for a ticker: request the missing data through REST
+		marketSummaryData, err := b.GetMarketSummary(tickerData.Symbol)
+		if err != nil {
+			return err
+		}
+
+		tickerPrice = b.constructTicker(tickerData, &marketSummaryData, pair, asset.Spot)
+		b.Websocket.DataHandler <- tickerPrice
+
+		return nil
+	}
+
+	tickerPrice.Last = tickerData.LastTradeRate
+	tickerPrice.Bid = tickerData.BidRate
+	tickerPrice.Ask = tickerData.AskRate
+
+	b.Websocket.DataHandler <- tickerPrice
+
+	return nil
+}
+
+// WsProcessUpdateMarketSummary processes an update on the ticker
+func (b *Bittrex) WsProcessUpdateMarketSummary(marketSummaryData *MarketSummaryData) error {
+	pair, err := currency.NewPairFromString(marketSummaryData.Symbol)
+	if err != nil {
+		return err
+	}
+
+	tickerPrice, err := ticker.GetTicker(b.Name, pair, asset.Spot)
+	if err != nil {
+		// Received partial data for a ticker: request the missing data through REST
+		var tickerData TickerData
+		tickerData, err = b.GetTicker(marketSummaryData.Symbol)
+		if err != nil {
+			return err
+		}
+
+		tickerPrice = b.constructTicker(tickerData, marketSummaryData, pair, asset.Spot)
+		b.Websocket.DataHandler <- tickerPrice
```

Comment:
here as well. Let me know what you think. I just don't like the potential of it impeding order book updates or account update speed.